src/Forms/RestoreAutoBackup.cs

```diff
@@ -166,30 +168,26 @@ private void linkLabelOpenContainingFolder_LinkClicked(object sender, LinkLabelL
 
         public static void CleanAutoBackupFolder(string autoBackupFolder, int autoBackupDeleteAfterMonths)
         {
-            const int maxCount = 100; // to avoid locking computer
-            if (Directory.Exists(autoBackupFolder))
+            lock (_locker) // only allow one thread
             {
-                var targetDate = DateTime.Now.AddMonths(-autoBackupDeleteAfterMonths);
-                var files = Directory.GetFiles(autoBackupFolder, "*.*");
-                int filesDeleted = 0;
-                foreach (string fileName in files)
+                if (Directory.Exists(autoBackupFolder))
                 {
-                    try
+                    var targetDate = DateTime.Now.AddMonths(-autoBackupDeleteAfterMonths);
+                    var files = Directory.GetFiles(autoBackupFolder, "*.*");
+                    foreach (string fileName in files)
                     {
-                        var name = Path.GetFileName(fileName);
-                        if (RegexFileNamePattern.IsMatch(name) && Convert.ToDateTime(name.Substring(0, 10), CultureInfo.InvariantCulture) < targetDate)
+                        try
                         {
-                            File.Delete(fileName);
-                            filesDeleted++;
-                            if (filesDeleted > maxCount)
+                            var name = Path.GetFileName(fileName);
+                            if (RegexFileNamePattern.IsMatch(name) && Convert.ToDateTime(name.Substring(0, 10), CultureInfo.InvariantCulture) <= targetDate)

```

Comment:
Here we want to delete backups after a specific month:
`Convert.ToDateTime(name.Substring(0, 10), CultureInfo.InvariantCulture) <= targetDate`

The old code was `Convert.ToDateTime(name.Substring(0, 10), CultureInfo.InvariantCulture) == targetDate` which I think is wrong...

Reply:
The old code was ´<´... (not `==`)

Reply:
> The old code was ´<´... (not `==`)
Exactly, typo